pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -196,6 +202,31 @@ func (i *Instance) IsTopLevelInstance() bool {
 	return !i.IsChildInstance()
 }
 
+const (
+	planInProgressReason   = "PlanInProgress"
+	resourceNotReadyReason = "ResourceNotReady"
+	readyConditionType     = "Ready"
+)
+
+func (i *Instance) SetReadiness(ready bool, msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionTrue, Message: msg}
+	if !ready {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = resourceNotReadyReason
+	}
+
+	meta.SetStatusCondition(&i.Status.Conditions, condition)
+}
+
+// SetReadinessUnknownForDeploy sets an unknown state for Status.Conditions.Ready
+// this will modify instance only when deploy, upgrade, update plan is running
+func (i *Instance) SetReadinessUnknownForDeploy() {
```

Comment:
Shouldn't _any_ running plan result in an unknown state?
Also a nit, because this isn't used only for `deploy`:

```suggestion
func (i *Instance) SetReadinessUnknown() {
```

Reply:
yeah my idea was for this to not be a dumb setter but also include the business logic around plans but I failed to find a good name :( so I made it a dumb setter :(